pkg/ipamd/ipamd.go

```diff
@@ -2371,3 +2302,31 @@ func (c *IPAMContext) isConfigValid() bool {
 
 	return true
 }
+
+func (c *IPAMContext) AddFeatureToCNINode(ctx context.Context, featureName rcv1alpha1.FeatureName, featureValue string) error {
+	cniNode := &rcv1alpha1.CNINode{}
+	key := types.NamespacedName{
+		Name:      c.myNodeName,
+		Namespace: "default",
+	}
+	err := c.cachedK8SClient.Get(ctx, key, cniNode)
+	if err != nil {
+		return err
+	}
+
+	var names []rcv1alpha1.FeatureName
+	for _, feature := range cniNode.Spec.Features {
```

Comment:
Check out: https://github.com/samber/lo#containsby

```
if lo.ContainsBy(cniNode.Spec.Features, func(feature rcv1alpha1.Feature) bool {
  return feature == rcv1alpha1.Feature{ Name: featureName, Value: featureValue }
}) {
  return nil
}
```

Reply:
nice lib. Will use this. Thanks